src/anniversary/sf33rd/Source/Game/EFFECT.c

```diff
@@ -0,0 +1,510 @@
+            break;
+        tmw = (WORK_Other *)frw[wk->wu.shell_ix[i]];
+        if (tmw->refrected)
+            continue;
+        if ((tmw->wu.original_vitality == 0x1F) || (tmw->wu.original_vitality == 0x21) ||
+            (tmw->wu.original_vitality == 0x2E)) {
+            return 1;
+        }
+    }
+    return 0;
+}
+
+s32 clear_caution_flag(PLW *wk, u8 /* unused */) {
+    wk->caution_flag = 0;
+    return 0;
+}
+
+s32 set_caution_flag(PLW *wk, u8 /* unused */) {
+    wk->caution_flag = 1;
+    return 0;
+}
+
+s32 setup_status_flag(WORK *wk, u8 data) {
+    wk->pat_status = data;
+    return 0;
+}
+
+s32 reset_extra_bg_flag(WORK *wk, u8 /* unused */) {
+    another_bg[wk->id] = 0;
+    return 0;
+}
+
+s32 flip_my_rl_flag(WORK *wk, u8 /* unused */) {
+    wk->rl_flag = wk->rl_flag + 1U & 1;
+    return 0;
+}
+
+s32 setup_meoshi_hit_flag(WORK *wk, u8 data) {
+    wk->meoshi_hit_flag = data;
+    return 0;
+}
+
+s32 exec_char_asxy(WORK *wk, u8 data) {
+    s16 *from_rom2;
+    s32 st;
+    s16 ix = data;
+
+        ix *= 2;
+    from_rom2 = &wk->step_xy_table[ix];
+    st = *from_rom2++;
+    st *= 256;
+
+    if (wk->rl_flag) {
+        wk->xyz[0].cal += st;
+    } else {
+        wk->xyz[0].cal -= st;
+    }
+
+    st = *from_rom2;
+    st *= 256;
+    wk->xyz[1].cal += st;
+    return 0;
+}
+
+s32 setup_my_clear_level(WORK *wk, u8 data) {
+    wk->my_clear_level = data;
+    return 0;
+}
+
+s32 setup_my_bright_level(WORK *wk, u8 data) {
+    wk->my_bright_level = data;
+    return 0;
+}
+
+s32 setup_free_program(s32 /* unused */, s32 /* unused */) {
+    return 0;
+}
+
+s32 setup_bg_quake_x(s32 /* unused */, u8 data) {
+    bg_w.quake_x_index = data;
+    return 0;
+}
+
+s32 setup_bg_quake_y(s32 /* unused */, u8 data) {
+	#if defined(TARGET_PS2)
+    void pp_screen_quake(s32);
+    #endif
+    bg_w.quake_y_index = data;
+    pp_screen_quake(bg_w.quake_y_index);
+    return 0;
+}
+
+s32 setup_exdm_ix(PLW *wk, u8 data) {
+    wk->exdm_ix = data;
+    return 0;
+}
+
+s32 setup_dmv_use_flag(PLW *wk, u8 data) {
+    wk->dm_vital_use = data;
+    return 0;
+}
+
+s32 setup_disp_flag(WORK *wk, u8 data) {
+    wk->disp_flag = data;
+    return 0;
+}
+
+s32 setup_command_number(PLW *wk, u8 data) {
+    wk->cmd_request = data;
+    return 0;
+}
+
+u8 old_my_char_check(u8 num, u8 flag) {
+    switch (flag) {
+    case 0:
+        if ((Country == 1) || (Country == 8)) {
+            if (num > 14) {
+                num += 1;
+            }
+        } else if (num > 13) {
+            num += 1;
+        }
+
+        break;
+    case 1:
+        if (num > 14) {
+            num += 1;
+        }
+        break;
+    }
+    return num;
+}
+
+void effect_work_kill_mod_plcol(void) {
+    effect_work_kill(6, -1);
+}
+
+void setup_shadow_of_the_Effy(WORK *wk) {
+    wk->kage_flag = 1;
+    wk->kage_hx = 0;
+    wk->kage_hy = -0xA;
+    wk->kage_prio = 0x47;
+    wk->kage_char = 0xC;
+}
+
+void set_init_A4_flag(void) {
+    plw->init_E3_flag = 1;
```

Comment:
Let's make indexing more consistent
```suggestion
    plw[0].init_E3_flag = 1;
```